backend/core/src/entity/application.entity.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  UpdateDateColumn,
+  ManyToOne, BaseEntity
+} from "typeorm"
+import { User } from "./user.entity"
+import { Listing } from "./listing.entity"
+import { IsJSON, IsString, IsUUID } from "class-validator"
+
+@Entity({ name: "applications" })
+export class Application extends BaseEntity {
+  @PrimaryGeneratedColumn("uuid")
+  @IsString()
+  @IsUUID()
+  id: string
+
+  @CreateDateColumn()
+  createdAt: Date
+
+  @UpdateDateColumn()
+  updatedAt: Date
+
+  @ManyToOne((type) => User, (user) => user.applications)
+  user: User
+
+  @Column()
+  @IsString()
+  userId: string
```

Comment:
It allows flat response object structure (one could achieve this with querybuilder and without new column too, but much more complicated), otherwise a response would contain a user object:

```
{
  ...
  user: {
    id: id
  }
},
```
 instead of 

```
{
  ...
  userId: id
}
```

I'm eager to hear comments about this approach.

Reply:
@bencpeters up